asmcomp/s390x/emit.mlp

```diff
@@ -662,11 +712,53 @@ let fundecl fundecl =
   emit_string code_space;
   `	.align	8\n`;
   `{emit_symbol fundecl.fun_name}:\n`;
+  cfi_startproc ();
+
+  (* Dynamic stack checking *)
+  let stack_threshold_size = Config.stack_threshold * 8 in (* bytes *)
+  let { max_frame_size; contains_nontail_calls} =
+    preproc_stack_check
+      ~fun_body:fundecl.fun_body ~frame_size:(frame_size env) ~trap_size:16
+  in
+  let handle_overflow = ref None in
+  if contains_nontail_calls || max_frame_size >= stack_threshold_size then begin
+    let overflow = new_label () and ret = new_label () in
+    let threshold_offset = Domainstate.stack_ctx_words * 8 + stack_threshold_size in
+    let f = max_frame_size + threshold_offset in
+    let offset = Domainstate.(idx_of_field Domain_current_stack) * 8 in
+    `  lg %r1, {emit_int offset}(%r10)\n`;
+    `  lay %r1, {emit_int f}(%r1)\n`;
+    `  clgr %r15, %r1\n`;
```

Comment:
It might be possible to shave one instruction by using a `clg` instruction (compare register with memory).  Something like (untested!):
```suggestion
    `  lay %r1, {emit_int (-f)}(%r15)\n`;
    `  clg %r1, {emit_int offset}(%r10)\n`;
```